src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -3284,71 +3309,392 @@ new_readdir(DIR *dirp)
+	char  *str_cont             = NULL;
+	char  *str_maxeq            = NULL;
+	char  *str_enforcement      = NULL;
+
+	if (envp == NULL) {
+		num_entry = 0;
+	} else if (envp[0] == NULL) {
+		num_entry = 0;
+	} else {
+		while (envp[num_entry]) {
+			if (strncmp(envp[num_entry], "LD_PRELOAD", 10) == 0) {
+				preload_included = true;
+				idx_preload      = num_entry;
+				if (strstr(envp[num_entry], "libpil4dfs.so"))
+					pil4dfs_in_preload = true;
+			} else if (strncmp(envp[num_entry], "D_IL_REPORT", 11) == 0) {
+				report_included = true;
+			} else if (strncmp(envp[num_entry], "DAOS_MOUNT_POINT", 16) == 0) {
+				mp_included = true;
+			} else if (strncmp(envp[num_entry], "DAOS_POOL", 9) == 0) {
+				pool_included = true;
+			} else if (strncmp(envp[num_entry], "DAOS_CONTAINER", 14) == 0) {
+				cont_included = true;
+			} else if (strncmp(envp[num_entry], "D_IL_MAX_EQ", 11) == 0) {
+				maxeq_included = true;
+			} else if (strncmp(envp[num_entry], "D_IL_ENFORCE_EXEC_ENV", 21) == 0) {
+				enforcement_included = true;
+			}
+			num_entry++;
+		}
+	}
+
+	/* The fd for daos logging may be not valid here, so use plain printf */
+	new_envp = malloc(sizeof(char *) * (num_entry + N_ENV_CHECK + 1));
+	if (new_envp == NULL) {
+		printf("Error: failed to allocate memory for new_envp. Use existing envp\n");
+		goto err_out0;
+	}
+
+	pil4df_path = query_pil4dfs_path();
+	len2        = strnlen(pil4df_path, PATH_MAX);
+	/* copy existing entries */
+	for (i = 0; i < num_entry; i++)	{
+		if (preload_included == true && pil4dfs_in_preload == false && idx_preload == i) {
+			/* need to replace the existing string */
+			len = strnlen(envp[i], MAX_ARG_STRLEN);
+			/* 2 for ':' and '\0' */
+			len_total = len + len2 + 2;
+			if (len_total > MAX_ARG_STRLEN) {
+				printf("Error: env for LD_PRELOAD is too long!\n");
+				goto err_out1;
+			}
+			rc = asprintf(&new_preload_str, "%s:%s", envp[i], pil4df_path);
+			if (rc < 0) {
+				printf("Error: failed to allocate memory for LD_PRELOAD env!\n");
+				goto err_out1;
+			}
+			new_envp[i] = new_preload_str;
+			continue;
+		}
+		new_envp[i] = envp[i];
+	}
+	if (preload_included == false && pil4dfs_in_preload == false) {
+		rc = asprintf(&new_preload_str, "LD_PRELOAD=%s", pil4df_path);
+		if (rc < 0) {
+			printf("Error: failed to allocate memory for LD_PRELOAD env!\n");
+			goto err_out1;
+		}
+		new_envp[i] = new_preload_str;
+		i++;
+	}
+	if (!report_included) {
+		if (report)
+			str_report = strndup("D_IL_REPORT=1", 14);
+		else
+			str_report = strndup("D_IL_REPORT=0", 14);
+		if (str_report == NULL) {
+			printf("Error: failed to allocate memory for D_IL_REPORT env!\n");
+			goto err_out2;
+		}
+		new_envp[i] = str_report;
+		i++;
+	}
+	if (!mp_included) {
+		fs_root = getenv("DAOS_MOUNT_POINT");
+		if (fs_root) {
+			rc = asprintf(&str_mp, "DAOS_MOUNT_POINT=%s", fs_root);
+			if (rc < 0) {
+				printf("Error: failed to allocate memory for DAOS_MOUNT_POINT "
+				       "env!\n");
```

Comment:
Applies elsewhere but please don't use ! in error messages!

It looks like this would all fit on line line as well.

Reply:
Thank you! Removed "!". What did you mean "all fit on line line"? Is it "all fit on one line"?

Reply:
You could remove the newline so the code could fit on one line of the source file rather than being wrapped onto 2.  We allow 100 characters per line.

Reply:
Thank you! I double-checked this line. There are still 101 characters in this line after "\n" is removed.